src/components/RecipeCard/RecipeCard.js

```diff
@@ -1,52 +1,30 @@
 import './RecipeCard.css';
 import {Link} from 'react-router-dom'
 import { getUser } from '../../APICalls.js'
+import { Component, Redirect } from 'react';
 
-function RecipeCard(props) {
-  //This whole card needs a thorough come over and I think we could be performing
+ let RecipeCard = (props) => {
+  //This whole card needs a thorough comb over and I think we could be performing
   //way less logic and actually just look at dynamic props
-  let route = `/recipepage/${props.id}`
 
+  let route = `/recipepage/${props.id}`
   let user = JSON.parse(localStorage.getItem('user'));
-  if (!user) user = { userRecipes: [] }
+  let buttonText, rating;
 
-  async function updateUser() {
-    //What is happening here?
-    let response = await getUser(user.id)
-    // localStorage.setItem('user', JSON.stringify(response.getInfo))
+  if(user && user.userRecipes.find(recipe => recipe.recipeId == props.id)){
+    buttonText = 'Already Donated'
+  } else if (user && props.userId === user.id) {
+    buttonText = 'Your recipe'
+  } else {
+    buttonText = "Give N' Get Recipe"
   }
 
-  let rating;
-  if (props.rating != null) {
+  if (props.rating !== null) {
     rating = props.rating + " out of 5";
   } else {
     rating = "Not yet rated";
   }
 
```

Comment:
Thanks for fixing this. I apologize for the spaghetti code. I know it was logic heavy and not idea but was trying to get it working.

Reply:
No worries at all! I am glad it was up and running and could comb through it later :)